### DISCUSSION TITLE
move from USB to Ethernet coordinator using raspberry?

### DISCUSSION THREAD

**Discussion by toxic0berliner** [Koenkk/zigbee2mqtt#16620] [Devices]
Hello,
I’m currently running z2m as a Hass plugin with the Sonoff USB dongle E with experimental support plugged directly into my hypervisor proxmox that is passing the USB port to my home assistant os VM. All works fine just fine 😁
I want high availability for HAss so I’m planning on migrating it to a proxmox cluster so the VM could run on any of my 3 hypervisor.
Issue is of course USB is not suited for that, not all 3 machines can be plugged at the same time to the Sonoff dongle.
I was looking at buying a network coordinator I have seen there is an open source compatible one for around $100 with POE, nice… But I also remembered that I have a raspberry pi 3 doing nothing in my closet. It would probably allow me to not repair all my devices (53 of them) if I could find something running on the raspberry to turn the USB into serial over Ethernet.
Do you know of something that would work with z2m that could do the trick for me or should I shell out for the supported Ethernet coordinator and re-pair all my devices and routers ?

Thanks in advance for any advice.

**Comment by Danit2** [comment]
Try this: 

https://www.zigbee2mqtt.io/advanced/remote-adapter/connect_to_a_remote_adapter.html

**Reply by toxic0berliner** [reply]
Wow, amazing, thanks a lot! Will have a try !
